### PR TITLE
[#697] Synchronize the latest local position between worker nodes.

### PR DESCRIPTION
**Pull request by Git-Yang**
#697 

## What is the purpose of the change
When the local position is merged with the synchronized data, it is impossible to check whether the synchronized data is the latest data, resulting in the situation that the new position is overwritten by the previous position.

In order to solve the above problems, we require that when each worker node synchronizes its position with other nodes, it synchronizes the updated position of the local task to ensure that other worker nodes get the latest position.

## Brief changelog

1. Config management service log print modification.
2. Record the updated position of the local task.
3. Only the updated position is synchronized, when synchronizing with other nodes.
4. The synchronized position is not merged to the local to avoid overwriting the latest local position, when the new node is online.

ps: Other nodes still synchronize the full position, when the new node is online.

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/service/OffsetManagementServiceImpl.java [hunk 1]
```diff
@@ -154,7 +178,6 @@ public void onCompletion(Throwable error, String key, Map<ByteBuffer, ByteBuffer
             boolean changed = false;
             switch (OffsetChangeEnum.valueOf(key)) {
                 case ONLINE_KEY:
-                    mergeOffsetInfo(result);
```
Comment (odbozhou) [inline rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/service/OffsetManagementServiceImpl.java:157]: How to synchronize data when starting ？

File: rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/service/PositionManagementServiceImpl.java [hunk 2]
```diff
@@ -154,7 +178,6 @@ public void onCompletion(Throwable error, String key, Map<ByteBuffer, ByteBuffer
             boolean changed = false;
             switch (PositionChangeEnum.valueOf(key)) {
                 case ONLINE_KEY:
-                    mergePositionInfo(result);
```
Comment (odbozhou) [inline rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/service/PositionManagementServiceImpl.java:157]: How to synchronize data when starting ？

Comment (Git-Yang) [reply rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/service/PositionManagementServiceImpl.java:157]: The offset of the starting node generally lags behind that of the existing node. If the node is restarted after being stopped for a period of time, direct merging may overwrite the latest offset, resulting in data duplication. 
   In addition, the offset of the starting node has been synchronized to the corresponding topic, and other nodes have actually obtained all the offsets, so I think it is not necessary to directly merge the offsets of the starting node.

File: rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/service/OffsetManagementServiceImpl.java (hunk 1 above)
Comment (Git-Yang) [reply rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/service/OffsetManagementServiceImpl.java:157]: The offset of the starting node generally lags behind that of the existing node. If the node is restarted after being stopped for a period of time, direct merging may overwrite the latest offset, resulting in data duplication.
In addition, the offset of the starting node has been synchronized to the corresponding topic, and other nodes have actually obtained all the offsets, so I think it is not necessary to directly merge the offsets of the starting node.

**Pull request merged**